### PR TITLE
Add Examples of the NGSIv2 normalized format for some data models

### PR DESCRIPTION
**Pull request by jmcanterafonseca**
@jason-fox r? 

This PR just adds examples of using the normalized format of NGSIv2 to some data models, as requested by the SynchroniCity project.

### CODE REVIEW

File: specs/Environment/AirQualityObserved/schema.json [hunk 1]
```diff
@@ -33,7 +33,10 @@
           "maximum": 1.0
         },
         "refDevice": {
-          "$ref": "https://fiware.github.io/dataModels/common-schema.json#/definitions/EntityIdentifierType"
+          "type": "array",
```
[the remark this answers is not in the document]
Comment (chicco785) [reply specs/Environment/AirQualityObserved/schema.json]: this also may "break" current working instances, should we have anyOf ?

File: specs/Environment/AirQualityObserved/doc/spec.md [hunk 2]
```diff
@@ -133,8 +133,112 @@ Below is the description of the attribute to be used for option A/.
...
+        }
+    }, 
+    "CO_Level": {
+        "value": "moderate"
+    }
+}
+```
     
-## Examples of use
+## Examples of use 2 (?options=keyValues simplified representation for data consumers)
```
[the remark this answers is not in the document]
Comment (jmcanterafonseca) [reply specs/Environment/AirQualityObserved/doc/spec.md]: agreed

File: specs/Environment/AirQualityObserved/schema.json (hunk 1 above)
[the remark this answers is not in the document]
Comment (jmcanterafonseca) [reply specs/Environment/AirQualityObserved/schema.json]: agreed revoking this change

File: specs/Parking/OffStreetParking/doc/spec.md [hunk 3]
```diff
@@ -308,7 +308,86 @@ all the zones.
...
+        "type": "DateTime", 
+        "value": "2016-06-02T09:25:55.00Z"
+    }, 
+    "description": {
+        "value": "Municipal car park located near the Trindade metro station and the Town Hall"
+    }
+}
+```
+
+## Examples of use 2
```
[the remark this answers is not in the document]
Comment (jmcanterafonseca) [reply specs/Parking/OffStreetParking/doc/spec.md]: for the moment we they need to co-exist. in the future maybe we need to cut and only have the normalized example. let's see

File: specs/Parking/OnStreetParking/doc/spec.md [hunk 4]
```diff
@@ -181,12 +181,109 @@ This value must aggregate free spots from all groups devoted to special parking
...
+}
+```
+
+## Examples of use 2
 
 An on street parking which contains a group of parking spots reserved for disabled people.
 At root entity level is announced that special parking spots for disabled are present and two of them free. 
 
-Main `OnstreetParking` entity. 
+Main `OnstreetParking entity.
```
[the remark this answers is not in the document]
Comment (jmcanterafonseca) [reply specs/Parking/OnStreetParking/doc/spec.md]: fixed

File: specs/Transportation/Vehicle/Vehicle/doc/spec.md [hunk 5]
```diff
@@ -225,4 +292,4 @@ T.B.D.
 ## Issues
 
 * Taxonomy of service types
-* What vehicle special usage categories are defined by the different countries? 
+* What vehicle special usage categories are defined by the different countries?
```
[the remark this answers is not in the document]
Comment (jmcanterafonseca) [reply specs/Transportation/Vehicle/Vehicle/doc/spec.md]: fixed

**Comment by jmcanterafonseca** [comment]
@jason-fox @chicco785 please could you r-review? 

We need to land this first so that it does not block further PRs.

File: specs/Environment/AirQualityObserved/doc/spec.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (jason-fox) [reply specs/Environment/AirQualityObserved/doc/spec.md]: Fine - NTC for this PR . Fix to be implemented in issue #291

File: .travis.yml [hunk 6]
```diff
@@ -25,7 +25,7 @@ matrix:
         - npm install eslint-plugin-json --save-dev -g
         - npm install -g jshint
       ## Data model validator
-        - npm install -g fiware-model-validator
```
Comment (chicco785) [inline .travis.yml]: - npm install -g -C ./validator

File: .travis.yml [hunk 7]
```diff
@@ -37,7 +37,7 @@ matrix:
       before_script:
         - eslint --ext .json .
         - jshint .
-        - validate -i common-schema.json -i geometry-schema.json -i specs/Weather/weather-schema.json -i specs/Alert/alert-schema.json -p . -c true
+        - node ./validator/validate.js -i common-schema.json -i geometry-schema.json -i specs/Weather/weather-schema.json -i specs/Alert/alert-schema.json -p . -c true
```
Comment (chicco785) [inline .travis.yml]: ```validate -i common-schema.json -i geometry-schema.json -i specs/Weather/weather-schema.json -i specs/Alert/alert-schema.json -p . -c true```

using install -g you can keep it unchanged

File: keyValues2Normalized.py [hunk 8]
```diff
@@ -0,0 +1,72 @@
+#!/usr/bin/python
```
Comment (chicco785) [inline keyValues2Normalized.py]: this code looks not used, i understand more or less the purpose, but i think it would be better in a separate PR

File: specs/Parking/ParkingAccess/example-normalized.json [hunk 9]
```diff
@@ -0,0 +1,31 @@
...
+    "refOffStreetParking": {
+        "type": "Relationship", 
+        "value": "porto-OffStreetParking-23889"
+    }, 
+    "features": {
+        "value": [
+            "barrier"
+        ]
+    }
+}
```
Comment (chicco785) [inline specs/Parking/ParkingAccess/example-normalized.json]: missing CR

File: specs/Parking/ParkingAccess/example.json [hunk 10]
```diff
@@ -0,0 +1,12 @@
...
+      "type": "ParkingAccess",
+      "name": "Trinidade main entrance",
+      "location": {
+        "coordinates": [-8.60961198807, 41.150691773],
+        "type": "Point"
+      },
+      "category": ["vehicleEntrance"],
+      "refOffStreetParking": "porto-OffStreetParking-23889",
+      "features": ["barrier"]
+}
```
Comment (chicco785) [inline specs/Parking/ParkingAccess/example.json]: Missing CR

File: specs/Parking/ParkingGroup/example-normalized.json [hunk 11]
```diff
@@ -0,0 +1,74 @@
...
+    }, 
+    "chargeType": {
+        "value": [
+            "free"
+        ]
+    }, 
+    "description": {
+        "value": "Two parking spots reserved for disabled people"
+    }
+}
```
Comment (chicco785) [inline specs/Parking/ParkingGroup/example-normalized.json]: Missing CR

File: specs/Parking/ParkingGroup/example.json [hunk 12]
```diff
@@ -0,0 +1,25 @@
...
+            [-3.803161973253841,43.46301091092682 ],
+            [-3.803147082548618,43.462879859445884],
+            [-3.803536474744068,43.462838666196674],
+            [-3.80356167695194, 43.46296641666926]
+          ]
+        ]
+      },
+      "requiredPermit": "disabledPermit",
+      "permitActiveHours": "null"
+}
```
Comment (chicco785) [inline specs/Parking/ParkingGroup/example.json]: Missing CR

File: specs/Parking/ParkingSpot/example.json [hunk 13]
```diff
@@ -0,0 +1,12 @@
...
+      "type": "ParkingSpot",
+      "name": "A-13",
+      "location": {
+        "type": "Point",
+        "coordinates": [-3.80356167695194, 43.46296641666926]
+      },
+      "status": "free",
+      "category": ["onstreet"],
+      "refParkingSite": "santander:daoiz_velarde_1_5"
+}
```
Comment (chicco785) [inline specs/Parking/ParkingSpot/example.json]: Missing CR

File: specs/PointOfInterest/PointOfInterest/example-normalized.json [hunk 14]
```diff
@@ -0,0 +1,41 @@
...
+            "addressCountry": "ES", 
+            "addressLocality": "Vilagarc\u00eda de Arousa"
+        }
+    }, 
+    "type": "PointOfInterest", 
+    "id": "PointOfInterest-A-Concha-123456", 
+    "name": {
+        "value": "Playa de a Concha"
+    }
+}
```
Comment (chicco785) [inline specs/PointOfInterest/PointOfInterest/example-normalized.json]: Missing CR

File: specs/Transportation/Bike/BikeHireDockingStation/example-normalized.json [hunk 15]
```diff
@@ -0,0 +1,37 @@
...
+    }, 
+    "address": {
+        "type": "PostalAddress", 
+        "value": {
+            "addressCountry": "ES", 
+            "addressLocality": "Barcelona", 
+            "streetAddress": "Gran Via Corts Catalanes,760"
+        }
+    }
+}
```
Comment (chicco785) [inline specs/Transportation/Bike/BikeHireDockingStation/example-normalized.json]: Missing CR

File: specs/Transportation/EVChargingStation/example-normalized.json [hunk 16]
```diff
@@ -0,0 +1,54 @@
...
+    }, 
+    "operator": {
+        "value": "Iberdrola"
+    }, 
+    "contactPoint": {
+        "value": {
+            "email": "vehiculoelectrico@ava.es"
+        }
+    }
+}
```
Comment (chicco785) [inline specs/Transportation/EVChargingStation/example-normalized.json]: Missing CR

File: specs/Transportation/TrafficFlowObserved/example-normalized.json [hunk 17]
```diff
@@ -0,0 +1,67 @@
...
+        "value": {
+            "addressLocality": "Valladolid", 
+            "addressCountry": "ES", 
+            "streetAddress": "Avenida de Salamanca"
+        }
+    }, 
+    "averageVehicleSpeed": {
+        "value": 52.6
+    }
+}
```
Comment (chicco785) [inline specs/Transportation/TrafficFlowObserved/example-normalized.json]: Missing CR

File: specs/Transportation/TrafficFlowObserved/example.json [hunk 18]
```diff
@@ -0,0 +1,34 @@
...
+       "dateObservedFrom": "2016-12-07T11:10:00Z",
+       "dateObservedTo": "2016-12-07T11:15:00Z",
+       "averageHeadwayTime": 0.5,
+       "intensity": 197,
+       "occupancy": 0.76,
+       "averageVehicleSpeed": 52.6,
+       "averageVehicleLength": 9.87,
+       "reversedLane": false,
+       "laneDirection": "forward"
+}
```
Comment (chicco785) [inline specs/Transportation/TrafficFlowObserved/example.json]: Missing CR

File: specs/Transportation/Vehicle/Vehicle/example-normalized.json [hunk 19]
```diff
@@ -0,0 +1,62 @@
...
+            }
+        }
+    },  
+    "serviceProvided": {
+        "value": [
+            "gargabeCollection", 
+            "wasteContainerCleaning"
+        ]
+    }
+}
```
Comment (chicco785) [inline specs/Transportation/Vehicle/Vehicle/example-normalized.json]: Missing CR

File: specs/Transportation/Vehicle/Vehicle/example.json [hunk 20]
```diff
@@ -0,0 +1,18 @@
...
+      },
+      "name": "C Recogida 1",
+      "speed": 50,
+      "cargoWeight": 314,
+      "serviceStatus": "onRoute",
+      "serviceProvided": ["garbageCollection", "wasteContainerCleaning"],
+      "areaServed": "Centro",
+      "refVehicleModel": "vehiclemodel:econic",
+      "vehiclePlateIdentifier": "3456ABC"
+}
```
Comment (chicco785) [inline specs/Transportation/Vehicle/Vehicle/example.json]: Missing CR

File: specs/Transportation/Vehicle/VehicleModel/example.json [hunk 21]
```diff
@@ -0,0 +1,11 @@
...
+      "id": "vehiclemodel:econic",
+      "type": "VehicleModel",
+      "name": "MBenz-Econic2014",
+      "brandName": "Mercedes Benz",
+      "modelName": "Econic",
+      "manufacturerName": "Daimler",
+      "vehicleType": "lorry",
+      "cargoVolume": 1000,
+      "fuelType": "diesel"
+}
```
Comment (chicco785) [inline specs/Transportation/Vehicle/VehicleModel/example.json]: Missing CR

File: specs/Transportation/Vehicle/VehicleModel/example-normalized.json [hunk 22]
```diff
@@ -0,0 +1,25 @@
...
+    "manufacturerName": {
+        "value": "Daimler"
+    }, 
+    "fuelType": {
+        "value": "diesel"
+    }, 
+    "vehicleType": {
+        "value": "lorry"
+    }
+}
```
Comment (chicco785) [inline specs/Transportation/Vehicle/VehicleModel/example-normalized.json]: Missing CR

File: specs/Environment/AirQualityObserved/example-normalized.json [hunk 23]
```diff
@@ -0,0 +1,74 @@
+{
+    "dateObserved": {
+        "value": "2016-03-15T11:00:00/2016-03-15T12:00:00"
+    }, 
+    "airQualityLevel": {
+        "value": "moderate"
+    }, 
+    "CO": {
+        "value": 500
```
Comment (chicco785) [inline specs/Environment/AirQualityObserved/example-normalized.json]: compared the doc, metadata are missing.

```
"metadata": {
             "unitCode": {
                 "value": "GP"
             }
         }
```

Comment (chicco785) [reply specs/Environment/AirQualityObserved/example-normalized.json]: same for the other measurements.

File: specs/Environment/NoiseLevelObserved/example-normalized.json [hunk 24]
```diff
@@ -0,0 +1,34 @@
...
+                42.8491
+            ]
+        }
+    }, 
+    "type": "NoiseLevelObserved", 
+    "id": "Vitoria-NoiseLevelObserved-2016-12-28T11:00:00_2016-12-28T12:00:00", 
+    "LAS": {
+        "value": 91.6
+    }
+}
```
Comment (chicco785) [inline specs/Environment/NoiseLevelObserved/example-normalized.json]: Missing CR

File: specs/Parking/OffStreetParking/example-normalized.json [hunk 25]
```diff
@@ -0,0 +1,74 @@
...
+        "value": "PT8H"
+    }, 
+    "dateModified": {
+        "type": "DateTime", 
+        "value": "2016-06-02T09:25:55.00Z"
+    }, 
+    "description": {
+        "value": "Municipal car park located near the Trindade metro station and the Town Hall"
+    }
+}
```
Comment (chicco785) [inline specs/Parking/OffStreetParking/example-normalized.json]: Missin CR

File: specs/Parking/OnStreetParking/example-normalized.json [hunk 26]
```diff
@@ -0,0 +1,92 @@
...
+    "availableSpotNumber": {
+        "value": 3,
+        "metadata": {
+            "timestamp": {
+                "value": "2018-09-12T12:00:00",
+                "type": "DateTime"
+            }
+        }
+    }
+}
```
Comment (chicco785) [inline specs/Parking/OnStreetParking/example-normalized.json]: Missing CR.

File: specs/Parking/OnStreetParking/example.json [hunk 27]
```diff
@@ -0,0 +1,30 @@
...
+            [-3.803161973253841,43.46301091092682 ],
+            [-3.803147082548618,43.462879859445884],
+            [-3.803536474744068,43.462838666196674],
+            [-3.80356167695194, 43.46296641666926]
+          ]
+        ]
+      },
+      "areaServed": "Zona Centro",
+      "refParkingGroup": ["daoiz-velarde-1-5-main", "daoiz-velarde-1-5-disabled"]
+  }
```
Comment (chicco785) [inline specs/Parking/OnStreetParking/example.json]: Missing CR.

File: validator/libs/checks.js [hunk 28]
```diff
@@ -175,71 +175,69 @@ module.exports = {
   //check if an example is supported by contextBroker
   exampleSupported: function(fullPath) {
     var check = true;
-    var apiInstance = getApiEntityClient();
+    var apiInstance=getApiEntityClient();
```
Comment (chicco785) [inline validator/libs/checks.js]: any reason to remove the spaces? if linter asked for that it looks strange.

File: validator/package-lock.json [hunk 29]
```diff
@@ -0,0 +1,793 @@
+{
```
Comment (chicco785) [inline validator/package-lock.json]: not 100% sure, but probably this can go in gitignore.

File: specs/Environment/AirQualityObserved/doc/spec.md [hunk 30]
```diff
@@ -133,8 +133,112 @@ Below is the description of the attribute to be used for option A/.
...
 mode (`options=keyValues`).
+
+## Examples of use 1 (Normalized Format)
+
+```json
+{
+    "id": "Madrid-AmbientObserved-28079004-2016-03-15T11:00:00",
+    "type": "AirQualityObserved",
+    "dateObserved": {
+        "value": "2016-03-15T11:00:00/2016-03-15T12:00:00"
```
Comment (chicco785) [inline specs/Environment/AirQualityObserved/doc/spec.md]: will the context broker return this or also the type?

File: keyValues2Normalized.py (hunk 8 above)
Comment (chicco785) [reply keyValues2Normalized.py]: also probably the simplest trick is the following:
load the keyValue example in contextBroker, query the model not using the keyValue, store the normalised example. Feelings?

File: .travis.yml (hunk 6 above)
Comment (jmcanterafonseca) [reply .travis.yml]: why do you want to install this globally? I think it is clearer as it is proposed, as we are ensuring that we are running the validator locally provided in the PR.

File: keyValues2Normalized.py (hunk 8 above)
Comment (jmcanterafonseca) [reply keyValues2Normalized.py]: The code has been used to generate the normalized examples, and then some manual tweaking. Using a Context Broker just for that purpose I think could be overkilling.

File: specs/Environment/AirQualityObserved/doc/spec.md (hunk 30 above)
Comment (jmcanterafonseca) [reply specs/Environment/AirQualityObserved/doc/spec.md]: we are conveying here just the Entity Representation that can be used by Context Producers to create the corresponding Entity.

File: specs/Transportation/Bike/BikeHireDockingStation/doc/spec.md [hunk 31]
```diff
@@ -149,7 +192,7 @@ Bike hire docking station real time data in Malaga
 
 ## Use it with a real service
 
-Ongoing implementation in Cedus project for Rennes and Málaga. Expected by 2017Q2.
+T.B.D.
```
[the remark this answers is not in the document]
Comment (jmcanterafonseca) [reply specs/Transportation/Bike/BikeHireDockingStation/doc/spec.md]: I don't know but it is better clear that statement.

File: validator/libs/checks.js (hunk 28 above)
Comment (jmcanterafonseca) [reply validator/libs/checks.js]: don't know the reason why it happened

File: .travis.yml (hunk 6 above)
Comment (jmcanterafonseca) [reply .travis.yml]: and we also show developers how to run the validator locally which it is rather important. 

A follow up PR could introduce npm scripts so that we just do npm test to validate the data models. that can be concurrent to the suggestion made by @jason-fox of introducing prettier in the code.

File: .travis.yml (hunk 7 above)
Comment (jmcanterafonseca) [reply .travis.yml]: see above

File: specs/Parking/OnStreetParking/example.json (hunk 27 above)
Comment (jmcanterafonseca) [reply specs/Parking/OnStreetParking/example.json]: all have been fixed

File: keyValues2Normalized.py (hunk 8 above)
Comment (chicco785) [reply keyValues2Normalized.py]: we use it anyhow for the validation, it would have been just an additional step ;)
